src/k8s/pkg/utils/pki/generate.go

```diff
@@ -22,7 +22,7 @@ func GenerateSerialNumber() (*big.Int, error) {
 	return serialNumber, nil
 }
 
-func GenerateCertificate(subject pkix.Name, years int, ca bool, dnsSANs []string, ipSANs []net.IP) (*x509.Certificate, error) {
+func GenerateCertificate(subject pkix.Name, notAfter time.Time, ca bool, dnsSANs []string, ipSANs []net.IP) (*x509.Certificate, error) {
```

Comment:
the notAfter depends on the current time, but a separate time.Now() is included in this function to do this. in this case, pass both notBefore and notAfter as parameters